File: developer/Samples/NET 4.6/WF_WCF_Samples/WCF/Basic/Binding/ByteStreamEncoder/CS/ByteStreamHttpBinding/TransferModeHelper.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

using System.ComponentModel;
using System.ServiceModel;

namespace Microsoft.Samples.ByteStreamEncoder
{

    static class TransferModeHelper
    {
        public static bool IsDefined(TransferMode v)
        {
            return ((v == TransferMode.Buffered) || (v == TransferMode.Streamed) ||
                (v == TransferMode.StreamedRequest) || (v == TransferMode.StreamedResponse));
        }

        public static bool IsRequestStreamed(TransferMode v)
        {
            return ((v == TransferMode.StreamedRequest) || (v == TransferMode.Streamed));
        }

        public static bool IsResponseStreamed(TransferMode v)
        {
            return ((v == TransferMode.StreamedResponse) || (v == TransferMode.Streamed));
        }

        public static void Validate(TransferMode value)
        {
            if (!IsDefined(value))
            {
                throw(new InvalidEnumArgumentException("value", (int)value,
                    typeof(TransferMode)));
            }
        }
    }
}

File: developer/Samples/NET 4.6/WF_WCF_Samples/WCF/Basic/Binding/ByteStreamEncoder/VB/ByteStreamHttpBinding/TransferModeHelper.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved.
'----------------------------------------------------------------

Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System.ServiceModel

Namespace Microsoft.Samples.ByteStreamEncoder

    Friend NotInheritable Class TransferModeHelper

        Private Sub New()
        End Sub

        Public Shared Function IsDefined(ByVal v As TransferMode) As Boolean
            Return ((v = TransferMode.Buffered) OrElse (v = TransferMode.Streamed) OrElse (v = TransferMode.StreamedRequest) OrElse (v = TransferMode.StreamedResponse))
        End Function

        Public Shared Function IsRequestStreamed(ByVal v As TransferMode) As Boolean
            Return ((v = TransferMode.StreamedRequest) OrElse (v = TransferMode.Streamed))
        End Function

        Public Shared Function IsResponseStreamed(ByVal v As TransferMode) As Boolean
            Return ((v = TransferMode.StreamedResponse) OrElse (v = TransferMode.Streamed))
        End Function

        Public Shared Sub Validate(ByVal value As TransferMode)
            If (Not IsDefined(value)) Then
                Throw (New InvalidEnumArgumentException("value", CInt(Fix(value)), GetType(TransferMode)))
            End If
        End Sub
    End Class
End Namespace
